### PR TITLE
JIT: Add support for struct returns from Swift reverse pinvokes

### PR DESCRIPTION
**Pull request by jakobbotsch**
Update the 10 tests to have some struct returns as well.

We also have to change the prestub to save `rax` since it's used as a register for the ret buffer in Swift calls.

### CODE REVIEW

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib PTAL @amanasifkhalid 

Failures look like #99810 and other known failures

FYI @jkurdek @kotlarmilos @matouskozak

File: src/coreclr/jit/lclvars.cpp [hunk 1]
```diff
@@ -169,6 +169,28 @@ void Compiler::lvaInitTypeRef()
...
+        const CORINFO_SWIFT_LOWERING* lowering   = GetSwiftLowering(retTypeHnd);
+        if (lowering->byReference)
+        {
+            printf("Swift compilation returns %s by reference\n", typGetObjLayout(retTypeHnd)->GetClassName());
+        }
+        else
+        {
+            printf("Swift compilation returns %s as %d primitive(s) in registers\n",
+                   typGetObjLayout(retTypeHnd)->GetClassName(), lowering->numLoweredElements);
+            for (size_t i = 0; i < lowering->numLoweredElements; i++)
```
Comment (amanasifkhalid) [inline src/coreclr/jit/lclvars.cpp:185]: I think we dump these lowered element types in a few places now, right? You may want to consider moving this logic to a helper method (though feel free to do that in a future PR).

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks!

File: src/coreclr/jit/lclvars.cpp (hunk 1 above)
Comment (jakobbotsch) [reply src/coreclr/jit/lclvars.cpp:185]: Yeah. Makes sense to factor this, let me do that in a follow-up.

**Comment by jakobbotsch** [comment]
Also FYI @jkoritzinsky and cc @jkotas for the stub change.

File: src/coreclr/vm/amd64/theprestubamd64.S [hunk 2]
```diff
@@ -6,17 +6,20 @@
 #include "asmconstants.h"
 
 NESTED_ENTRY ThePreStub, _TEXT, NoHandler
-        PROLOG_WITH_TRANSITION_BLOCK 0, 0, 0, 0, 0
+        PROLOG_WITH_TRANSITION_BLOCK 8, 0, 0, 0, 0
+        mov             [rsp + __PWTB_FloatArgumentRegisters - 8], rax
```
Comment (jkotas) [inline src/coreclr/vm/amd64/theprestubamd64.S:10]:
```suggestion
        mov             [rsp + __PWTB_FloatArgumentRegisters - 8], rax // Return buffer in Swift calling convention
```

File: src/coreclr/vm/amd64/theprestubamd64.S [hunk 3]
```diff
@@ -6,17 +6,20 @@
...
 
         //
         // call PreStubWorker
         //
         lea             rdi, [rsp + __PWTB_TransitionBlock]     // pTransitionBlock*
         mov             rsi, METHODDESC_REGISTER
         call            C_FUNC(PreStubWorker)
+        mov             r10, rax
 
+        mov             rax, [rsp + __PWTB_FloatArgumentRegisters - 8]
```
Comment (jkotas) [inline src/coreclr/vm/amd64/theprestubamd64.S:20]:
```suggestion
        mov             rax, [rsp + SIZEOF_MAX_OUTGOING_ARGUMENT_HOMES]
```

File: src/coreclr/vm/amd64/theprestubamd64.S (hunk 2 above)
Comment (jkotas) [reply src/coreclr/vm/amd64/theprestubamd64.S:10]: We use `SIZEOF_MAX_OUTGOING_ARGUMENT_HOMES` to refer to the extra locals in the other place where they are used:

Comment (jkotas) [reply src/coreclr/vm/amd64/theprestubamd64.S:10]: Ok, I looked into a wrong file. I would just use [rsp] then.

**Comment by jkotas** [comment]
The stub change LGTM

**Pull request merged**